x-pack/filebeat/input/s3/input.go

```diff
@@ -440,16 +440,30 @@ func (p *s3Input) newS3BucketReader(svc s3iface.ClientAPI, s3Info s3Info) (*bufi
 		return nil, errors.New("s3 get object response body is empty")
 	}
 
+	// Check content-type
+	if resp.ContentType != nil {
+		switch *resp.ContentType {
+		case "text/plain", "text/plain; charset=utf-8":
+			return bufio.NewReader(resp.Body), nil
```

Comment:
Isn't it the same as default case?

Reply:
yeah, good catch. I will remove this case section! Thanks!